app/components/public/ticket-list.js

```diff
@@ -99,6 +99,21 @@ export default Component.extend(FormMixin, {
     return input;
   }),
 
+  didInsertElement() {
+    this.data.forEach(ticket => {
+      ticket.set('discount', 0);
+    });
+    if (this.code) {
+      this.send('togglePromotionalCode', this.code);
+    }
+    if (this.tickets.length === 1) {
+      const preSelect = Math.max(this.tickets[0].minOrder, 1);
+      this.tickets[0].set('orderQuantity', preSelect);
+      this.order.tickets.addObject(this.tickets[0]);
+      this.send('updateOrderAmount');
+    }
+  },
```

Comment:
Why did you move it up?

Reply:
U didn' told me to move this code above togglePromotionalCode?

Reply:
Above `this.send('togglePromotionalCode')`